Breeding-Insight/bi-api
# [BI-1363] Share Ontology

# Description
**Story:** https://breedinginsight.atlassian.net/jira/software/c/projects/BI/boards/1?modal=detail&selectedIssue=BI-1363

- Added an OntologyController for sharing, revoke, and retrieving shared ontologies. 
- Added tests
- Scoped out some TODOs where dependencies prevented me from working further

# Dependencies
biweb: BI-1363

# Testing
1. Create multiple programs with the same species
2. Create an additional program of a different species
3. Go to the Program Management -> Configuration page
4. Make sure the programs with same species are able to be shared and program with different species is not. 
5. Share with a program or two
6. Modal should automatically close and the ontologies should be listed under 'Shared Ontology' now
7. The shared programs should have a `(Not Accepted)` next to them (accepting share not in this card)
8. Revoke the share from a program or two and make sure those are removed from the list of shared programs. 

# Checklist:
